### DISCUSSION TITLE
Develop tree node linkage

### DISCUSSION THREAD

**Discussion by JohnDev297** [community/community#47185] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm developing a horizontal tree node using Vue 3 (TypeScript) and Quasar Framework and facing some issue on the linking lines between parent node and child nodes. Here is an example when parent node connect to third child node the linkage is break where the first and second node height will grow automatically depends on how many sibling it has, so I would like to know how can I perform some sort of calculation to achieve this kind of design?

**Comment by 2MuchC0ff33** [comment]
Cool question @JohnDev297 

Reading through your code, it appears you're using a fixed height for the parent and child nodes.

One potential solution is to use CSS flexbox to adjust the nodes' height based on their content dynamically.

Here's how you can update the styles to implement this solution:
```
<template>
  <template v-for="node in state.tree_nodes" :key="node.id">
    <div
      :class="node.node_type != 'starting' ? 'node-group ' : ''"
    >
      <div
        ref="nodeRefs"
        :class="
          'node row no-wrap relative-position ' +
          (node.next_has_true ? 'node-next-has-true ' : '') +
          (node.next_has_false ? 'node-next-has-false ' : '') +
          (node.node_type == 'true'
            ? 'node-prev-is-true '
            : node.node_type == 'false'
            ? 'node-prev-is-false '
            : node.node_type == 'next'
            ? 'node-prev-is-true '
            : '')
        "
      >
        <div class="node-parent">
          <q-card class="node-card">
            <q-card-section>{{ node.node_type }}</q-card-section>

            <div class="green-start-line"></div>
            <div class="red-start-line"></div>
            <div class="yellow-start-line"></div>
            <div class="blue-start-line"></div>
          </q-card>
        </div>
        <div class="node-children" style="display: flex; flex-direction: column;">
          <tree-node :nodes="node.children" />
        </div>
      </div>
    </div>
  </template>
</template>

```

The updated script wraps each node in a container with 'display: flex' and sets 'flex-direction: column' to allow the child nodes' height to grow dynamically.

**Reply by JohnDev297** [reply]
But still can't solve the broken linkage.

**Reply by 2MuchC0ff33** [reply]
Hrrm.. you could try absolute positioning for the lines, with the starting and ending points calculated based on the parent and child node positions. To calculate the positions dynamically, you'll need to use JavaScript or a CSS preprocessor like SASS or LESS. For example, you could use JavaScript to calculate the position of the parent node and then use that information to determine where the linking lines should begin. The position of the child node could be used to calculate the ending position of the linking lines. Alternatively, you could use CSS Grid Layout to create a grid with the parent and child nodes and then position the linking lines using grid lines.

Hopefully, more people will reply to this thread and provide additional assistance with either of these solutions!

**Reply by JohnDev297** [reply]
Hmmm I get what you mean let me have a try.

**Comment by JohnDev297** [comment, marked as answer]
Solved by using offsetTop
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop
